packages/jest/src/plugins/plugin.ts

```diff
@@ -459,3 +581,75 @@ function requireJestUtil<T>(
     paths: [dirname(resolvedJestCorePaths[jestPath])],
   }));
 }
+
+async function getJestOption<T = any>(
+  rawConfig: any,
+  absConfigFilePath: string,
+  optionName: string
+): Promise<T> {
+  if (rawConfig[optionName]) return rawConfig[optionName];
+
+  if (rawConfig.preset) {
+    const dir = dirname(absConfigFilePath);
+    const presetPath = resolve(dir, rawConfig.preset);
+    try {
+      const preset = await loadConfigFile(presetPath);
```

Comment:
We should cache the preset. This will be called multiple times, and most of the time, it will load the same preset path.

Reply:
I feel like we should do this at the `loadConfigFile` level. It can lead to unexpected caching issues if we're not careful. For example, we need to bust the cache if either the preset changes, or anything it import changes.